reconcile/utils/saasherder/interfaces.py

```diff
@@ -50,6 +50,10 @@ def secret_parameters(self) -> SaasSecretParameters:
 class SaasApp(Protocol):
     name: str
 
+    @property
+    def parent_app(self) -> Optional[SaasApp]:
+        ...
+
```

Comment:
i think the mypy errors you get are related to this protocol. it would need to match `AppV1`and `AppV1_AppV1`. while `AppV1` provides all the things the protocol needs, `AppV1_AppV1` does not (lacks the `parent_app`).

Reply:
maybe something like this could work? (disclaimer: i did not test this)

```
class SaasApp(Protocol):
    @property
    def name(self) -> str:
        ...

    @property
    def parent_app(self) -> Optional[Union[SaasApp, SaasAppWithoutParent]]:
        ...


class SaasAppWithoutParent(Protocol):
    @property
    def name(self) -> str:
        ...

Reply:
ah i see you solved it already

Reply:
@chassing is telling me what to do behind the scenes while i am ranting